engine/src/cmd/jump_capable.cpp

```diff
@@ -0,0 +1,580 @@
+                        // expensive ones should counter planets...this is safe now, for gameplay
+                        shiphack *= fabs( unit->specInterdiction );
+                }
+            }
+            float   multipliertemp = 1;
+            float   minsizeeffect  = (planet->rSize() > smallwarphack) ? planet->rSize() : smallwarphack;
+            float   effectiverad   = minsizeeffect*( 1.0f+UniverseUtil::getPlanetRadiusPercent() )+unit->rSize();
+            if (effectiverad > bigwarphack)
+                effectiverad = bigwarphack;
+            QVector dir     = unit->Position()-planet->Position();
+            double  udist   = dir.Magnitude();
+            float   sigdist = UnitUtil::getSignificantDistance( unit, planet );
+            if ( planet->isPlanet() && udist < (1<<28) ) //If distance is viable as a float approximation and it's an actual celestial body
+                udist = sigdist;
+            do {
+                double dist = udist;
+                if (dist < 0) dist = 0;
+                dist *= shiphack;
+                if ( dist > (effectiverad+warpregion0) )
+                    multipliertemp = pow( (dist-effectiverad-warpregion0), curvedegree )*upcurvek;
+                else
+                    multipliertemp = 1;
+                if (multipliertemp < minmultiplier) {
+                    minmultiplier = multipliertemp;
+                    *nearest_unit = planet;
+                    //eventually use new multiplier to compute
+                } else {break; }
+            } while (0);
+            if (!testthis)
+                break; //don't want the ++
+        }
+    }
+    return minmultiplier;
+}
+
+
+float JumpCapable::CourseDeviation( const Vector &OriginalCourse, const Vector &FinalCourse ) const
+{
+    const Unit *unit = static_cast<const Unit*>(this);
+    if (unit->ViewComputerData().max_ab_speed() > .001)
+        return ( OriginalCourse-(FinalCourse) ).Magnitude()/unit->ViewComputerData().max_ab_speed();
+    else
+        return (FinalCourse-OriginalCourse).Magnitude();
+}
+
+
+void JumpCapable::DeactivateJumpDrive()
+{
+    Unit *unit = static_cast<Unit*>(this);
+    if (unit->jump.drive >= 0)
+        unit->jump.drive = -1;
+}
+
+
+const std::vector< std::string >& JumpCapable::GetDestinations() const
+{
+    const Unit *unit = static_cast<const Unit*>(this);
+    return unit->pImage->destination;
+}
+
+
+const Energetic::UnitJump& JumpCapable::GetJumpStatus() const
+{
+    const Unit *unit = static_cast<const Unit*>(this);
+    return unit->jump;
+}
+
+
+StarSystem* JumpCapable::getStarSystem()
+{
+    Unit *unit = static_cast<Unit*>(this);
+    if (activeStarSystem) {
+        return activeStarSystem;
+    } else {
+        Cockpit *cp = _Universe->isPlayerStarship( unit );
+        if (cp)
+            if (cp->activeStarSystem)
+                return cp->activeStarSystem;
```

Comment:
Braces needed

Reply:
I'll do a round of adding braces. However, I should note that this isn't a deep refactor of the code. I merely moved existing code to a new file.
If I were to try and also refactor every line in it, it would have taken me far more than a few hours.